Do-ItCore/Organization/NotificationManager.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  NotificationManager.swift
+//  Do-ItCore
+//
+//  Created by Cesar F. Chacon on 2/27/19.
+//  Copyright © 2019 The Swifter Picker-Uppers. All rights reserved.
+//
+
+import UserNotifications
+
+class NotificationManager {
+    let center = UNUserNotificationCenter.current()
+
+    func requestAuthorization() {
+        center.requestAuthorization(options: [.alert, .sound]) {
+            (granted, error) in
+                // do something if granted is false, user did not allow notifications
+        }
+    }
+
+    func setTrigger(_ doIt: DoIt) {
+        let content = UNMutableNotificationContent()
+        content.title = "Your DoIt is due " + doIt.dueDate.description
+        content.body = "Go finish it"
+        let dateComponents = Calendar.current.dateComponents([.year, .month, .day, .hour, .minute, .second], from: doIt.dueDate)
+        let trigger = UNCalendarNotificationTrigger(dateMatching: dateComponents, repeats: false)
```

Comment:
We want the notification to be delivered before the do-it is due, not at the moment of its due date. We're allowing the user to decide how long prior to the due date the notification should be sent, so you can add a Date parameter to the `setTrigger` function and use that date to determine when the notification should be delivered.

That Date parameter is also what should be used for determing the body text. You can you get the string using DateComponentsFormatter.